code/modules/unit_tests/explosion_action.dm

```diff
@@ -0,0 +1,214 @@
+	run_loc_floor_bottom_left.ChangeTurf(original_open_turf_type, original_open_baseturfs)
+	run_loc_floor_top_right.ChangeTurf(original_closed_turf_type, original_closed_baseturfs)
+
+#undef OPEN_FLOOR_TYPE
+#undef CLOSED_FLOOR_TYPE
+
+/// From the base implementation of `/atom`, [var/max_integrity].
+#define MAX_INTEGRITY_VALUE 500
+
+/// Tests the `EX_ACT()` macro on objs to ensure some level of the underlying framework still functions.
+/datum/unit_test/explosion_action/proc/execute_obj_tests()
+	// we're using the abstract type here because we don't need anything stronger for this test.
+	var/obj/test_object = allocate(/obj)
+
+	test_object.update_integrity(MAX_INTEGRITY_VALUE)
+
+	EX_ACT(test_object, EXPLODE_NONE)
+	TEST_ASSERT_EQUAL(test_object.get_integrity(), MAX_INTEGRITY_VALUE, "EX_ACT() with EXPLODE_NONE severity should not have altered the integrity of the target, but instead saw a change!")
+	test_object.update_integrity(MAX_INTEGRITY_VALUE) // just here for cleanliness
+
+	EX_ACT(test_object, EXPLODE_LIGHT) // can do anywhere from 10 to 90 damage, let's just care if it's not equal or not
+	TEST_ASSERT_NOTEQUAL(test_object.get_integrity(), MAX_INTEGRITY_VALUE, "EX_ACT() with EXPLODE_LIGHT severity should have altered the integrity of the target, but instead saw no change!")
+	test_object.update_integrity(MAX_INTEGRITY_VALUE)
+
+	EX_ACT(test_object, EXPLODE_HEAVY) // can do anywhere from 100 to 250 damage
+	TEST_ASSERT_NOTEQUAL(test_object.get_integrity(), MAX_INTEGRITY_VALUE, "EX_ACT() with EXPLODE_HEAVY severity should have altered the integrity of the target, but instead saw no change!")
+	test_object.update_integrity(MAX_INTEGRITY_VALUE)
+
+	EX_ACT(test_object, EXPLODE_DEVASTATE) // does an INFINITE amount of damage, will trigger a qdel()
+	TEST_ASSERT(QDELETED(test_object), "EX_ACT() with EXPLODE_DEVASTATE severity should have deleted the target, but instead saw no change!")
+
+#undef MAX_INTEGRITY_VALUE
+
+/// Sets up a fully armored corgi for testing purposes. Split out into its own proc as to not clutter up the main test.
+/datum/unit_test/explosion_action/proc/set_up_test_dog()
+	var/mob/living/basic/pet/dog/corgi/returnable_dog = allocate(/mob/living/basic/pet/dog/corgi)
+	returnable_dog.maxHealth = MAX_LIVING_HEALTH
+	returnable_dog.health = MAX_LIVING_HEALTH
+
+	var/obj/item/clothing/head/helmet/invincible_hat = allocate(/obj/item/clothing/head/helmet)
+	invincible_hat.set_armor(/datum/armor/immune)
+	returnable_dog.inventory_head = invincible_hat
+
+	var/obj/item/clothing/suit/armor/vest/invincible_vest = allocate(/obj/item/clothing/suit/armor/vest)
+	invincible_vest.set_armor(/datum/armor/immune)
+	returnable_dog.inventory_back = invincible_vest
+
+	return returnable_dog
+
+/// Proc to lessen the amount of copypasta we do for the alien tests, simply sets the rolling vars we have.
+/datum/unit_test/explosion_action/proc/get_alien_damages(mob/living/carbon/alien/subject)
```

Comment:
Nit: `read_alien_damages` or something might be nicer because `get_` implies a return value